service/src/main/scala/org/apache/celeborn/server/common/service/config/FsConfigServiceImpl.java

```diff
@@ -0,0 +1,125 @@
+
+public class FsConfigServiceImpl implements ConfigService {
+  private static final Logger LOG = LoggerFactory.getLogger(FsConfigServiceImpl.class);
+  private CelebornConf celebornConf;
+  private final AtomicReference<SystemConfig> systemConfigAtomicReference = new AtomicReference<>();
+  private final AtomicReference<Map<String, TenantConfig>> tenantConfigAtomicReference = new AtomicReference<>(new HashMap<>());
+  private static final String CONF_TENANT_ID = "tenantId";
+  private static final String CONF_LEVEL = "level";
+  private static final String CONF_CONFIG = "config";
+
+  private final ScheduledExecutorService configRefreshService =
+      ThreadUtils.newDaemonSingleThreadScheduledExecutor("config-refresh-service");
+
+  public FsConfigServiceImpl(CelebornConf celebornConf) {
+    this.celebornConf = celebornConf;
+    this.refresh();
+    long dynamicConfigRefreshTime = celebornConf.dynamicConfigRefreshTime();
+    this.configRefreshService.scheduleWithFixedDelay(
+        () -> refresh(),
+        dynamicConfigRefreshTime,
+        dynamicConfigRefreshTime,
+        TimeUnit.MILLISECONDS);
+  }
+
+  private synchronized void refresh() {
+    File configurationFile = getConfigurationFile(System.getenv());
+    if (!configurationFile.exists()) {
+      return;
+    }
+
+    SystemConfig systemConfig = null;
+    Map<String, TenantConfig> tenantConfs = new HashMap<>();
+    try (FileInputStream fileInputStream = new FileInputStream(configurationFile)) {
+      Yaml yaml = new Yaml();
+      List<Map<String, Object>> dynamicConfigs = yaml.load(fileInputStream);
+      for (Map<String, Object> settings : dynamicConfigs) {
+        String tenantId = (String) settings.get(CONF_TENANT_ID);
+        String level = (String) settings.get(CONF_LEVEL);
+        Map<String, String> config =
+            ((Map<String, Object>) settings.get(CONF_CONFIG))
+                .entrySet().stream()
+                .collect(Collectors.toMap(a -> a.getKey(), a -> a.getValue().toString()));
+        if (ConfigLevel.TENANT.name().equals(level)) {
+          TenantConfig tenantConfig = new TenantConfig(this, tenantId, config);
+          tenantConfs.put(tenantId, tenantConfig);
+        } else {
+          systemConfig = new SystemConfig(celebornConf, config);
+        }
+      }
+    } catch (Exception e) {
+      LOG.warn("Refresh dynamic config error: {}", e.getMessage(), e);
+    }
+
+    tenantConfigAtomicReference.set(tenantConfs);
+    systemConfigAtomicReference.set(systemConfig == null ? new SystemConfig(celebornConf) : systemConfig);
+  }
+
+  @Override
+  public SystemConfig getSystemConfig() {
+    return systemConfigAtomicReference.get();
+  }
+
+  @Override
+  public TenantConfig getRawTenantConfig(String tenantId) {
+    return tenantConfigAtomicReference.get().get(tenantId);
+  }
+
+  @Override
+  public void refreshAllCache() {
+    this.refresh();
+  }
+
+  @Override
+  public void shutdown() {
+    configRefreshService.shutdown();
```

Comment:
Does this need to add timeout for shutdown? Like `ThreadUtils.shutdown( configRefreshService, 800.millis)`